Index: qca-wifi-g833d781-dirty/cmn_dev/hif/src/ce/ce_assignment.h
===================================================================
--- qca-wifi-g833d781-dirty.orig/cmn_dev/hif/src/ce/ce_assignment.h
+++ qca-wifi-g833d781-dirty/cmn_dev/hif/src/ce/ce_assignment.h
@@ -334,7 +334,7 @@ static struct CE_attr host_ce_config_wla
 	{ /* CE7 */ (CE_ATTR_DIAG_FLAGS | CE_ATTR_DISABLE_INTR),
 		0, 2, DIAG_TRANSFER_LIMIT, 2, NULL,},
 };
-
+#define PIPEDIR_INOUT_H2H 4  /* bidirectional, host to host */
 static struct CE_pipe_config target_ce_config_wlan[] = {
 	/* host->target HTC control and raw streams */
 	{ /* CE0 */ 0, PIPEDIR_OUT, 32,  256, CE_ATTR_FLAGS, 0,},
Index: qca-wifi-g833d781-dirty/os/linux/include/ieee80211_ioctl.h
===================================================================
--- qca-wifi-g833d781-dirty.orig/os/linux/include/ieee80211_ioctl.h
+++ qca-wifi-g833d781-dirty/os/linux/include/ieee80211_ioctl.h
@@ -25,6 +25,11 @@
 #ifndef _NET80211_IEEE80211_IOCTL_H_
 #define _NET80211_IEEE80211_IOCTL_H_
 
+  typedef unsigned char uint8_t;
+  typedef unsigned short uint16_t;
+  //typedef unsigned long uint32_t;
+  typedef unsigned long long uint64_t;
+
 /*
  * IEEE 802.11 ioctls.
  */
